nimare/generate.py

```diff
@@ -0,0 +1,267 @@
+"""Utilities for generating data for testing"""
+import os
+
+import numpy as np
+import nibabel as nib
+
+from .dataset import Dataset
+from .utils import get_resource_path
+from .meta.utils import compute_ma, get_ale_kernel
+from .transforms import vox2mm, mm2vox
+
+DEF_TINY = 1e-50
+
+
+def create_coordinate_dataset(
+    foci_num,
+    fwhm,
+    sample_size,
+    studies,
+    foci_coords=None,
+    foci_noise=0,
+    foci_weights=None,
+    seed=42,
+    sample_size_variance=None,
+):
+    """Generate coordinate based dataset for meta analysis.
+
+    Parameters
+    ----------
+    foci_num : :obj:`int` or :obj:`list`
+        The number of foci to be generated per study
+    fwhm : :obj:`float`
+        Full width at half maximum to define the probability
+        spread of the foci.
+    sample_size : :obj:`int` or :obj:`list`
+        Either mean number of participants in each study
+        or a list specifying the sample size for each
+        study.
+    studies : :obj:`int`
+        Number of studies to generate.
+    foci_coords : :obj:`list` of three item array_like objects (Optional)
+        Ground truth coordinates of foci.
+        If not specified, coordinates will be randomly selected.
+        If not the same length as the number of foci (i.e., `foci_num`),
+        foci will either be over-sampled or under-sampled.
+    foci_noise : :obj:`int` or :obj:`list`
+        Number of foci considered to be noise in each study
+        or a list of integers representing how many noise focis
+        there should be in each study.
+    foci_weights : :obj:`list` (Optional)
+        Weighing of each foci representing the probability of
+        that foci being sampled in a study.
+    seed : :obj:`int` (Optional)
+        Random state to reproducibly initialize random numbers.
+    sample_size_variance : :obj:`int` (Optional)
+        Variance of the number of participants in each study.
+        (must be >=0).
+
+    Returns
+    -------
+    ground_truth_foci : :obj:`list`
+        generated foci in xyz (mm) coordinates
+    dataset : :class:`nimare.Dataset`
+    """
+    rng = np.random.RandomState(seed=seed)
+    if isinstance(sample_size, list):
+        sample_sizes = sample_size
+        if len(sample_sizes) != studies:
+            raise ValueError("sample_size must be the same length as studies")
+    else:
+        if sample_size_variance is None:
+            raise ValueError("sample_size_variance must be specified")
+        sample_size_lower_limit = int(sample_size - sample_size_variance)
+        # add 1 since randint upper limit is a closed interval
+        sample_size_upper_limit = int(sample_size + sample_size_variance + 1)
+        sample_sizes = rng.randint(sample_size_lower_limit, sample_size_upper_limit, size=studies)
```

Comment:
This is a little confusing. Based on the variable name and description, I assumed that `sample_sizes` would be drawn from a normal distribution (or at least one with some definition of "variance").

Reply:
Just to be clear- I don't mean you should switch to a Gaussian distribution. Maybe just change the variable name or description to make it clearer that it's half the range of a uniform distribution.

Reply:
thanks for catching this, I switched the name to `sample_size_interval` and tried to give a more detailed description, how does that sound?

Reply:
That sounds great!